tests/unit/lms/models/lti_params_test.py

```diff
@@ -46,3 +46,20 @@ def test_it_doesnt_set_partial_keys(self):
         assert params["custom_canvas_course_id"] == "SOME_ID"
         # Nonexistent ones in the same "level" are not present
         assert "custom_canvas_api_domain" not in params
+
+    def test_prefers_lti1p1_ids(self):
+        params = LTIParams.from_v13(
+            {
+                "sub": "user_id-v13",
+                "https://purl.imsglobal.org/spec/lti/claim/resource_link_id": {
+                    "id": "resource_link_id-v13"
+                },
+                "https://purl.imsglobal.org/spec/lti/claim/lti1p1": {
+                    "user_id": "user_id-v11",
+                    "resource_link_id": "resource_link_id-v11",
+                },
+            }
+        )
+
+        assert params["user_id"] == "user_id-v11"
+        assert params["resource_link_id"] == "resource_link_id-v11"
```

Comment:
It's a little counter intuitive that the items which come later in the list are higher priority, but it's simple enough if you spot it (we overwrite). I don't think it's an issue if you test for it though as you'll get told off it you change it wrong.

How short could you get this test with parameterisation? If it's not too much different, then it would be nice to not mix the two.

Reply:
I've tried this a couple of ways, and I'm not really sure. Some observations:

 * The way it gets spaced out in a `parameterize` is obnoxious
 * Importing and interpolating `CLAIM_PREFIX` helps quite a bit
 * If you only do one case at a time, you can get away with just `v11` / `v13` as the value
 * If there were more cases, it might make more sense to parameterize
 
The parameterized version looks like this:

```python
    @pytest.mark.parametrize(
        "key,params",
        (
            (
                "user",
                {
                    "sub": "v13",
                    f"{CLAIM_PREFIX}/lti1p1": {"user_id": "v11"},
                },
            ),
            (
                "resource_link_id",
                {
                    f"{CLAIM_PREFIX}/resource_link_id": {"id": "v13"},
                    f"{CLAIM_PREFIX}/lti1p1": {"resource_link_id": "v11"},
                },
            ),
        ),
    )
    def test_prefers_lti1p1_idas(self, key, params):
        params = LTIParams.from_v13(params)

        assert params[key] == "v11"
```

This might be a case where reptition is actuallty better:

```python
    def test_prefers_lti1p1_resource_link_id(self):
        params = LTIParams.from_v13(
            {
                f"{CLAIM_PREFIX}/resource_link_id": {"id": "v13"},
                f"{CLAIM_PREFIX}/lti1p1": {"resource_link_id": "v11"},
            }
        )

        assert params["resource_link_id"] == "v11"

  # ... etc.
```